modules/ztp-nbde-troubleshooting-permanent-error-conditions.adoc

```diff
@@ -0,0 +1,62 @@
+// Module included in the following assemblies:
+//
+// scalability_and_performance/ztp-nbde-implementation-guide.adoc
+
+[id="ztp-nbde-troubleshooting-permanent-error-conditions_{context}"]
+= Troubleshooting permanent error conditions
+
+If the `READY` count does not equal the `DESIRED` count after an extended period of time, it might indicate a permanent failure condition. In this case, the following conditions might apply:
+
+* A typo in the Tang server URL or thumbprint in the `NEW_TANG_PIN` definition.
+* The Tang server is decommissioned or the keys are permanently lost.
+
+.Procedure
+
+. Validate the Tang server configuration by performing a simple encrypt and decrypt operation on each Tang server’s configuration as defined in the daemonset. For example:
++
+[source,terminal]
+----
+$ echo "okay" | clevis encrypt tang \
+  '{"url":"http://tangserver02:7500","thp":"badtumbprint"}' | \
+  clevis decrypt
+Unable to fetch advertisement: 'http://tangserver02:7500/adv/badthumbprint'!
+$ echo "okay" | clevis encrypt tang \
+  '{"url":"http://tangserver03:7500","thp":"goodthumbprint"}' | \
+  clevis decrypt
+okay
```

Comment:
This section is made up of 2 commands and 2 output blocks in a single block.  It should be split to match the other commands in the document.  I'd suggest something like this:

> Example of an encryption and decryption attempt with a bad thumbprint:
> ```
> $ echo "okay" | clevis encrypt tang \
>   '{"url":"http://tangserver02:7500","thp":"badtumbprint"}' | \
>   clevis decrypt
> ```
> Example output:
> ```
> Unable to fetch advertisement: 'http://tangserver02:7500/adv/badthumbprint'!
> ```
> Example of an encryption and decryption attempt with a good thumbprint:
> ```
> $ echo "okay" | clevis encrypt tang \
>   '{"url":"http://tangserver03:7500","thp":"goodthumbprint"}' | \
>   clevis decrypt
> ```
> Example output
> ```okay```

Reply:
changed